homeassistant/components/plugwise/climate.py

```diff
@@ -91,23 +86,43 @@ def hvac_mode(self) -> HVACMode:
     def hvac_action(self) -> HVACAction:
         """Return the current running hvac operation if supported."""
         # When control_state is present, prefer this data
-        if "control_state" in self.device:
-            if self.device.get("control_state") == "cooling":
-                return HVACAction.COOLING
-            # Support preheating state as heating, until preheating is added as a separate state
-            if self.device.get("control_state") in ["heating", "preheating"]:
-                return HVACAction.HEATING
-        else:
-            heater_central_data = self.coordinator.data.devices[
-                self.coordinator.data.gateway["heater_id"]
-            ]
-            if heater_central_data["binary_sensors"].get("heating_state"):
-                return HVACAction.HEATING
-            if heater_central_data["binary_sensors"].get("cooling_state"):
-                return HVACAction.COOLING
+        control_state: str = self.device.get("control_state", "not_found")
+        if control_state == "cooling":
+            return HVACAction.COOLING
+        # Support preheating state as heating, until preheating is added as a separate state
+        if control_state in ["heating", "preheating"]:
+            return HVACAction.HEATING
+        if control_state == "off":
+            return HVACAction.IDLE
+
+        heater_central_data = self.devices[self.gateway["heater_id"]]
+        if heater_central_data["binary_sensors"]["heating_state"]:
+            return HVACAction.HEATING
+        if heater_central_data["binary_sensors"].get("cooling_state", False):
+            return HVACAction.COOLING
 
         return HVACAction.IDLE
 
+    @property
+    def hvac_modes(self) -> list[HVACMode]:
+        """Return the current hvac modes."""
+        hvac_modes = [HVACMode.HEAT]
+        if self.gateway["cooling_present"]:
+            if self.gateway["smile_name"] == "Anna":
+                hvac_modes.append(HVACMode.HEAT_COOL)
```

Comment:
`HEAT_COOL`? As far as I'm aware, Anna doesn't support heating/cooling against temperature ranges and has a single setpoint?

If `HEAT_COOL` is used, I would expect this PR to also implement the `target_temperature_high` and `target_temperature_low` properties to support that.

Reply:
An Anna connected to a heat-pump supports automatic switching between heating and cooling.
The automation is based on two setpoints, one for switching to cooling, the other for switching back to heating mode.

Sorry, I don't know about these `target_*` properties, I would need to research them.

Reply:
HEAT_COOL is a temperature range in Home Assistant. The user sets a min/max and the climate device decides.

For example:

![image](https://user-images.githubusercontent.com/195327/170103950-29eaa8a1-e406-433c-b39b-8725bc477415.png)

Reply:
> An Anna connected to a heat-pump supports automatic switching between heating and cooling.

Hmm... I think it lands in my head and sees how you got here. I wonder how it would behave in that case. Basically we don't know the mode, but the HVAC action does reflect the correct one.

I'm not sure if we have a limitation on using heat_cool without having a range to be honest. I need to look into this.

Reply:
The present cooling-related climate code is partly based on assumptions, based on conversation with users that have actual cooling capability.
I have the impression that the Plugwise app/website will show one setpoint, only the heating or the cooling setpoint. Both in the Adam and the Anna cases. But I'm not sure.
I will ask my contact at Plugwise for the Anna-case.

Reply:
About the `target_temp_low / _high`, the presets all have a heating and a cooling setpoint. So they can be used as values for these two parameters.
The problem is the custom-option in the Plugwise schedule, that one allows you to set a single setpoint.
Fortunately, it's possible to change the preset setpoints.

So when a schedule is created with only presets then it would work with these two parameters.

More later today...

Reply:
@frenck  Is this a stopping issue? Or can you accept the present solution for now?

Thinking ahead, if we are to implement the `target_temp_low / _high` properties then first the backend needs some updating.

Reply:
I've made the change to separate HVACMode.HEAT and HVACMode.COOL modes, no need to update the backend I realized.

Reply:
After sleeping on this I realized this doesn't work: a user can now e.g. try changing the `hvac_mode` from HEAT to COOL and that will not work. So HVACMode.HEAT_COOL it must be.

How do I go about this in the backend? Do I let it output `target_temp`, target_temp_low` and `target_temp_high` and make the appropriate ones `None` when they are not used? And in `climate.py` have them all 3 as properties?